adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -935,6 +976,20 @@ private void setResponsePayload(final HttpServerResponse response, final Command
         }
     }
 
+    /**
+     * Validate if a command is valid and can be sent as response.
+     * <p>
+     * The default implementation will call {@link Command#isValid()}. Protocol adapters may override this, but should
+     * consider calling the super method.
+     * 
+     * @param command The command to validate, will never be {@code null}.
+     * @param currentSpan The current tracing span.
+     * @return {@code true} if the command is valid, {@code false} otherwise.
+     */
+    protected boolean isCommandValid(final Command command, final Span currentSpan) {
+        return command.isValid();
```

Comment:
are you going to also log the outcome of the check to the span?

Reply:
I simply refactored that method for now, but that could be an option. However, doing that here would be wrong place. As others, overriding this method, would need to actually do that, when they call into the super method here.